presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationCompiler.java

```diff
@@ -151,7 +152,7 @@ public static BindableAggregationFunction generateBindableAggregationFunction(Cl
                     }
 
                     genericSignature = genericSignature.isPresent() ? genericSignature : Optional.of(implementation.getSignature());
-                    // TODO validateSignature(signature, implementation.getSignature());
+                    validateSignaturesCompatibility(genericSignature, implementation.getSignature());
```

Comment:
If the previous line were rewritten as I suggested before, the two lines here would read:
```
if (!genericSignature.isPresent()) {
  genericSignature = Optional.of(implementation.getSignature());
}
validateSignaturesCompatibility(genericSignature, implementation.getSignature());
```

Notice that if the last line is put into an `else` branch, the logic here will be easier to follow.